### PR TITLE
Avoid write files if not development environment

### LINKED ISSUE #2146: Error when deploy to Google App Engine
**Informations**
- **Node.js version**: v10.0.0
- **npm version**: v6.0.0
- **Strapi version**: v3.0.0-alpha.16
- **Database**: MongoDB
- **Operating system**: at Google Cloud

**What is the current behavior?**

After deploy the app:

```
textPayload:  "Error: Cannot find module 'immutable'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/srv/plugins/content-type-builder/services/ContentTypeBuilder.js:7:31)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)" 
```

screenshot 2018-10-15 at 16 51 03

**Steps to reproduce the problem**

Deploy the app and go to https://project-id.appspot.com/

**What is the expected behavior?**

Working directly.

### PR DESCRIPTION
**Pull request by abdonrd**
My PR is a:
- [ ] 💥 Breaking change
- [x] 🐛 Bug fix #2146
- [ ] 💅 Enhancement
- [ ] 🚀 New feature

Main update on the:
- [ ] Admin
- [ ] Documentation
- [x] Framework
- [x] Plugin

### CODE REVIEW

File: packages/strapi/lib/core/plugins.js [hunk 1]
```diff
@@ -88,6 +88,10 @@ module.exports = async function() {
     return;
   }
 
+  if (strapi.config.environment !== 'development') {
```
Comment (soupette) [inline packages/strapi/lib/core/plugins.js]: Maybe we can add a check and a warning in the server if the file is missing. So the user would know that the front-end cannot work without the `plugins.json` file?

Comment (abdonrd) [reply packages/strapi/lib/core/plugins.js]: Something like this https://github.com/strapi/strapi/pull/2518/commits/7d039644fae62e63994546783fade20158db72ac?

**Comment by abdonrd** [comment]
@soupette @lauriejim could you review it? I would like to solve #2146 as soon as possible. Thanks in advance!

**Comment by lauriejim** [comment]
Hello @abdonrd!
Did you try with a `try/catch` instead of a condition ?

**Comment by abdonrd** [comment]
@lauriejim no. Do you prefer that I change it to `try/catch`?

**Comment by lauriejim** [comment]
After discussion with @Aurelsicoko:

When you have 3 environments `development`, `staging` and `production` we can't know if `production` env configs are for **production** server. 

In fact our 3 names are just like 3 packages of variables. It also could be `jedi`, `clones`, `vador`.

So to define you start your app in your **production** server you will to start with `--production` flag.
With that we don't care of the environment name you use.

Does it make sense ?
If yes, I will make the update of your PR to make it work like explain in my message.

**Comment by abdonrd** [comment]
@lauriejim ok! It also makes sense to me! 👍

**Comment by lauriejim** [comment]
So in this case to start your app you will have to run:
- `strapi start` it will start `development` env by default like now
- `NODE_ENV=staging npm start --production` Will use you `staging` env config. But in fact it's a **production** server (staging === production normally for your servers)
- `NODE_ENV=production npm start --production`

So all works :)

**Comment by abdonrd** [comment]
@lauriejim exactly! 😁

Thanks Jim!

**Comment by derrickmehaffy** [comment]
@abdonrd @lauriejim I would like to argue for the reverse here and say that this type of variable should be stored in the config itself to allow for modifications. As with a system like this, it basically defeats the purpose of having that config environment. We should try to cut down on the arguments required to start the server as this makes actual deployment much harder with something like PM2.

**Comment by lauriejim** [comment]
Yap I'm agree! 
Some changes, you will have to define in your environment if it's `production` or not.

**Comment by abdonrd** [comment]
Awesome, thanks @lauriejim & @derrickmehaffy!

There's some way I can try this before the release? To confirm that this is enough to solve #2146.

**Comment by derrickmehaffy** [comment]
Yup @lauriejim that looks good to me. @abdonrd you can fork the branch and build it from source.

On linux just make sure you have the `libpng-dev` package installed or you'll hit an error on build. Clone the source (make sure you have the global strapi package removed) then run `npm run setup:build` then create a dev project `strapi new test --dev`

**Comment by abdonrd** [comment]
> @abdonrd you can fork the branch and build it from source.
> 
> On linux just make sure you have the `libpng-dev` package installed or you'll hit an error on build. Clone the source (make sure you have the global strapi package removed) then run `npm run setup:build` then create a dev project `strapi new test --dev`

@lauriejim @derrickmehaffy but as we edit the `packages/strapi/lib/core/plugins.js`...
When the deploy server (GAE, Heroku, etc) runs the `npm install` will not have that change, right?

**Comment by derrickmehaffy** [comment]
mmmm not yet @abdonrd it would be difficult to test from sources

**Comment by abdonrd** [comment]
@lauriejim any option to create a small test release about this? I don't know if it would be something simple or would involve a lot of work.

**Comment by derrickmehaffy** [comment]
Would have to be pushed to NPM, you might be able to spin up your own private npm source. :thinking:

**Comment by lauriejim** [comment]
Hummm something you can try is to follow this article to release new version of Strapi in your local machine - https://blog.strapi.io/testing-npm-package-before-releasing-it-using-verdaccio-and-ngrok/

And then in your deploy process your are changing the `npm` registry to your local machine (via ngrok proxy).

That can be a solution to test it.

**Comment by abdonrd** [comment]
It seems that we have forgotten something:

https://github.com/strapi/strapi/issues/2146#issuecomment-454768366

**Comment by abdonrd** [comment]
@lauriejim @derrickmehaffy It seems that we have forgotten something: https://github.com/strapi/strapi/issues/2146#issuecomment-454768366

**Pull request merged**